src/miniscript/satisfy.rs

```diff
@@ -503,6 +503,14 @@ pub struct Satisfaction {
 
 impl Satisfaction {
     fn minimum(sat1: Self, sat2: Self) -> Self {
+        // If there is only one available satisfaction, we must choose that
+        // regardless of has_sig marker.
+        // This handles the case where both are unavailable.
+        match (&sat1.stack, &sat2.stack) {
```

Comment:
as discussed on the phone, this allows too much; need to split `Unavailable` to distinguish between "we won't satisfy this" and "we can't satisfy this" and only skip the hassig check in the "can't" case

Reply:
This makes sense, but it goes against our assumptions for Miniscript. From Pieter's website.

> The attacker only has access to hash preimages that honest users have access to as well. This is a reasonable assumption because hash preimages are revealed once globally, and then available to everyone.

Reply:
Based on this I think it is reasonable to have the previous commit as-is. I will be opening a separate PR for it in case there is more discussion to be had about it.

Reply:
I never really agreed with that paragraph, but in any case, it does not preclude some honest users from having access to hash preimages that others don't.

Reply:
Resolving, moving discussion to #167